lib/index.ts

```diff
@@ -1,28 +1,99 @@
-// convoluted example to show different compilation targets
-// are respected
-export default function(a: number, b: number): number {
-  const id = (x: any) => x
-  return id(a) + id(b)
-}
+import atxHeading from './atx-heading'
+import block from './block'
+import * as c from './characters'
+import paragraph from './paragraph'
 
-// examples below to check if the correct libs are loaded up
-// in your respective editors
+type ContextType = 'atxHeading' | 'paragraph' | 'block'
 
-// array.includes is from lib.es2016
-export function includes<T>(v: T, arr: T[]): boolean {
-  return arr.includes(v)
-}
+export class Tokenizer {
+  public data = ''
+  public line = 1
+  public column = 1
+  public virtualColumn = 1
+  public offset = 0
+  public tabSize = 2
 
-// padStart and padEnd are from lib.es2017
-export function padString(inputString: string, maxLength: number, fillWith?: string): string {
-  const strLen = inputString.length
+  public context: ContextType = 'block'
+  public stateHandlers: any
+  public contextInfo = {}
+  public state: string = ''
 
-  return inputString
-    .padStart(strLen + Math.floor(maxLength / 2), fillWith)
-    .padEnd(strLen + maxLength, fillWith)
-}
+  public block: any = block
```

Comment:
I'd recommend exporting `interface`s that show the shape of `block`, `atxHeading` and `paragraph` from the respective files and using that as the type here instead of `any`